src/redprl/lcf.sml

```diff
@@ -3,72 +3,43 @@ structure LcfLanguage = LcfAbtLanguage (RedPrlAbt)
 structure Lcf :
 sig
   include LCF_UTIL
-  val prettyState : jdg state -> PP.doc
+  val prettyState : jdg state -> Fpp.doc
 end =
 struct
   structure Lcf = Lcf (LcfLanguage)
   structure Def = LcfUtilPure (structure Lcf = Lcf structure J = RedPrlJudgment)
   open Def Lcf
   infix |> ||
 
+  (* TODO: clean up all this stuff with vsep *)
 
-  val prettySyms =
-    PP.text o ListSpine.pretty (fn (u, sigma) => Sym.toString u ^ " : " ^ RedPrlAbt.O.Ar.Vl.PS.toString sigma) ", "
-
-  val prettyVars = 
-    PP.text o ListSpine.pretty (fn (x, tau) => Var.toString x ^ " : " ^ RedPrlAbt.O.Ar.Vl.S.toString tau) ", "
+  fun @@ (f, x) = f x
+  infixr 0 @@
 
   fun prettyGoal (x, jdg) =
-    PP.concat
-      [PP.text "Goal ",
-        PP.text (RedPrlAbt.Metavar.toString x),
-        PP.text ".",
-        PP.nest 2 (PP.concat [PP.line, RedPrlSequent.pretty TermPrinter.toString jdg]),
-        PP.line]
-
+    Fpp.nest 2 @@ 
+      Fpp.vsep 
+        [Fpp.seq [Fpp.hsep [Fpp.text "Goal", Fpp.text (Metavar.toString x)], Fpp.text "."],
+         RedPrlSequent.pretty TermPrinter.ppTerm jdg]
 
-  val prettyGoals : jdg eff Tl.telescope -> {doc : PP.doc, env : J.env, idx : int} =
+  val prettyGoals : jdg Tl.telescope -> {doc : Fpp.doc, env : J.env, idx : int} = 
     let
       open RedPrlAbt
     in
-      Tl.foldl
+      Tl.foldl 
```

Comment:
Try [appl.whitespace](https://git-scm.com/docs/git-config#git-config-applywhitespace) and [core.whitespace](https://git-scm.com/docs/git-config#git-config-corewhitespace).